Tests/Common/IsolatorLimitResultProviderTests.cs

```diff
@@ -140,7 +143,7 @@ public void ConsumesMultipleMinutes()
             Assert.IsTrue(provider.Invocations.TrueForAll(invoc => invoc == 1));
 
             // give time to the monitor to register the time consumer ended
-            Thread.Sleep(50);
+            _timeMonitor.RemoveAll();
```

Comment:
I think force calling `RemoveAll` here removes the purpose of the test assert, because it's asserting the production implementation is removing those consumers that ended, maybe this can be achieved in some other way?